Sources/SwiftCrossUI/Views/Toggle.swift

```diff
@@ -1,5 +1,82 @@
-/// A control that is either on or off.
-public struct Toggle: ElementaryView, View {
+
+public struct Toggle: View {
+    /// The style of toggle shown.
+    public let selectedToggleStyle: ToggleStyle
+    /// The label to be shown on or beside the toggle.
+    var label: String
+    /// Whether the toggle is active or not.
+    var active: Binding<Bool>
+
+    /// Creates a toggle that displays a custom label.
+    public init(selectedToggleStyle: ToggleStyle = .button, _ label: String, active: Binding<Bool>) {
+        self.selectedToggleStyle = selectedToggleStyle
+        self.label = label
+        self.active = active
+    }
+
+    public var body: some View {
+        switch selectedToggleStyle {
+            case .switch:
+                HStack() {
+                    Text(label)
+                    ToggleSwitch(active: active)
+                }
+            case .button:
+                ToggleButton(label, active: active)
+    }
+  }
+}
+
+extension Toggle {
+    /// Sets the style of the toggle.
+    public func toggleStyle(_ selectedToggleStyle: ToggleStyle) -> Toggle {
+        return Toggle(
+            selectedToggleStyle: selectedToggleStyle, 
+            self.label, 
+            active: self.active
+        )
+    }
+}
+
+public enum ToggleStyle {
+    case `switch`
+    case button
+}
+
+/// A light switch style control that is either on or off.
+private struct ToggleSwitch: ElementaryView, View {
+    /// Whether the switch is active or not.
+    private var active: Binding<Bool>
```

Comment:
Similarly, since it's an internal implementation detail you can get rid of all the access level modifiers (`private`, `public`, etc) on its properties, inits, and methods.